x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/aggregate/TopHits.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.sql.expression.function.aggregate;
+
+import org.elasticsearch.xpack.sql.analysis.index.MappingException;
+import org.elasticsearch.xpack.sql.expression.Expression;
+import org.elasticsearch.xpack.sql.expression.Expressions;
+import org.elasticsearch.xpack.sql.expression.FieldAttribute;
+import org.elasticsearch.xpack.sql.tree.Source;
+import org.elasticsearch.xpack.sql.type.DataType;
+
+import java.util.Collections;
+
+import static org.elasticsearch.common.logging.LoggerMessageFormat.format;
+
+public abstract class TopHits extends AggregateFunction {
+
+    TopHits(Source source, Expression field, Expression sortField) {
+        super(source, field, sortField != null ? Collections.singletonList(sortField) : Collections.emptyList());
+    }
+
+    public Expression sortField() {
+        return parameters().isEmpty() ? null : parameters().get(0);
+    }
+
+    public DataType sortFieldDataType() {
```

Comment:
I don't think this method is needed - one can always say `sortField().dataType()`.

Reply:
I did that to avoid the caller checking if the `sortField()` is null.

Reply:
All expressions return only one datatype - of their result, adding a new method is confusing. It's simple to do the null check - in fact, it's is mandatory for sorting since one cannot just add a null, null directive (if the sort field would be null).